devito/operator.py

```diff
@@ -121,89 +122,62 @@ def __init__(self, expressions, **kwargs):
         # Track the DLE output, as it might be useful at execution time
         self._dle_state = dle_state
 
+        self.symbolic_data = [i for i in parameters if isinstance(i, SymbolicData)]
+        dims = [i for i in parameters if isinstance(i, Dimension)]
+
+        d_parents = [d.parent for d in dims if hasattr(d, 'parent')]
```

Comment:
Need not be just buffered. There's an entire class of derived `Dimension` objects that are being envisioned for the near future. Time subsampling , for example.